Fusetools 2564 - cherry picked

bfitzpat opened the pull request:
# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.


## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

bfitzpat commented:
Testing locally in a debug Eclipse instance and installing the built update site into a new Photon instance works great. Though we still have some test failures (locally) with the project templates, @lhein was able to work us through the issue with the local build. #happydance 

I will continue to test while we're in code freeze, but I think we can finally count this as done!

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/pom.xml:
```diff
@@ -1,228 +1,200 @@
+		</plugins>
+	</build>
+
+	<!-- Make sure that all these repos are saved in Nexus before we do the 
```
these repositories are now saved, comment can be removed

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest/META-INF/MANIFEST.MF:
```diff
@@ -11,40 +11,77 @@ Require-Bundle: org.fusesource.ide.foundation.core,
- lib/wsdl2rest-impl.jar,
- lib/wsdl2rest-util.jar,
- lib/commons-validator-1.6.jar
+ libs/args4j.jar,
```
why choosing to remove all the versions?
I see benefits that when upgrading versions, it will require less maintenance, I see downside that in the packaged plugin we don't have anymore an easy view of which version is embedded.
Other reasons?

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/.gitignore:
```diff
@@ -0,0 +1 @@
+lib/
```
I think that there is no more lib folder in this plugin so this .gitignore file can be removed

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,176 @@
+	 * @param wsdlURL
+	 */
+	public void setWsdlURL(String wsdlURL) {
+        String oldValue = this.wsdlURL;
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,176 @@
+	 * @param projectName
+	 */
+	public void setProjectName(String projectName) {
+        String oldValue = this.projectName;
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,176 @@
+	 * @param destinationJava
+	 */
+	public void setDestinationJava(String destinationJava) {
+        String oldValue = this.destinationJava;
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,176 @@
+	 * @param destinationCamel
+	 */
+	public void setDestinationCamel(String destinationCamel) {
+        String oldValue = this.destinationCamel;
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,176 @@
+	 * @param targetServiceAddress
+	 */
+	public void setTargetServiceAddress(String targetServiceAddress) {
+        String oldValue = this.targetServiceAddress;
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestOptions.java:
```diff
@@ -0,0 +1,176 @@
+	 * @param targetServiceAddress
+	 */
+	public void setTargetRestServiceAddress(String targetRestServiceAddress) {
+        String oldValue = this.targetRestServiceAddress;
```
indentation issue

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,359 @@
+		if (javaFile != null) {
+			ClassLoader oldLoader = Thread.currentThread().getContextClassLoader();
+			try {
+				// initialize bus using bundle classloader, to prevent project dependencies from leaking in
```
I still don't understand this comment.
Which bus is initialized?
Why "project dependencies from leaking in"?

bfitzpat replied:
Was copied from the Switchyard wrapper when I created this plugin. Can remove it.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,359 @@
+		}
+
+		// Gets native File using EFS
+		fileFound = EFS.getStore(uri).toLocalFile(0, new NullProgressMonitor());			
```
optional: would be best to pass a progress monitor instead of using a NullProgressMonitor, ti will allow to provide a smoother progress reporting to user

apupier replied:
I created a more generic task for taking care of Progress Monitor https://issues.jboss.org/browse/FUSETOOLS-2968

apupier replied:
with a proposal to be tested https://github.com/apupier/jbosstools-fuse/tree/FUSETOOLS-2968-useProgressMonitor

bfitzpat replied:
ok. something to explore once the first PR is in

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,359 @@
+		if (flag && !this.inTest && 
+				getContainer().getCurrentPage().equals(this.getStartingPage()) && 
+				pageTwo.getMessageType() == IMessageProvider.WARNING &&
+				pageTwo.getMessage().startsWith("Caution:")) { //$NON-NLS-1$
```
is it "Caution:" also if someone is using another language?
Where does this "Caution" coming from?

bfitzpat replied:
Caution comes from https://github.com/jbosstools/jbosstools-fuse/pull/1362/files#diff-9afd4b5273a020f133f09dce67f592d0 in one of the two "Caution:" messages.  It was an easy way of flagging a warning that didn't prevent the user from stopping the process, but does make them go to the next page in the wizard so they SEE those warnings. 

I'm open to suggestions, but this was a low tech solution. 

Considering that I don't know that we have actually localized anything though we have the facility to do so, I wasn't too concerned with what "Caution" might be in other languages.

bfitzpat replied:
Created https://issues.jboss.org/browse/FUSETOOLS-2970

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardBasePage.java:
```diff
@@ -0,0 +1,288 @@
+	/**
+	 * Shared Databinding Context for the page.
+	 */
+	protected final DataBindingContext dbc = new DataBindingContext(DisplayRealm.getRealm(Display.getCurrent()));
```
I'm surprised that you need to provide a specific Realm to the DataBindingContext. There was error with the default constructor?

bfitzpat replied:
Stuck with a tried and true method I'd used before with SwitchYard. If we don't need it, awesome.

bfitzpat replied:
works fine without it locally

apupier commented on editor/tests/pom.xml:
```diff
@@ -26,6 +26,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
 		<module>org.fusesource.ide.project.tests.integration</module>
 		<module>org.fusesource.ide.projecttemplates.tests</module>
 		<module>org.fusesource.ide.projecttemplates.tests.integration</module>
+        <module>org.fusesource.ide.wsdl2rest.ui.tests.integration</module>
```
indentation issue

apupier commented:
I wasn't able to have the generated route working with defautl configuration after the generation has been finished:
- with Spring project I have a build failure waiting for service to be available
- with SpringBoot project, I have the other route deployed but the rest route is not deployed.

I think that either, we need to modify the project in some way, either we need to tell user that some manual steps are still required.

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,359 @@
+	 */
+	protected IProject getSelectedProjectFromSelectionService() {
+		ISelectionService ss = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getSelectionService();
+		String projExpID = "org.eclipse.ui.navigator.ProjectExplorer"; //$NON-NLS-1$
```
using the Project explorer means that we are providing the menu only on "project explorer view"? Not in "Package explorer" for instance?
https://github.com/jbosstools/jbosstools-fuse/pull/1138#discussion_r180425769

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,359 @@
+				new org.fusesource.ide.camel.model.service.core.catalog.Dependency();
+		one.setArtifactId("jboss-jaxrs-api_2.0_spec"); //$NON-NLS-1$
+		one.setGroupId("org.jboss.spec.javax.ws.rs"); //$NON-NLS-1$
+		one.setVersion("1.0.0.Final-redhat-1"); //$NON-NLS-1$
```
where this version comes from? How it will evolve? Should we plan in a following iteration to have it configurable from the online "configuration" files depending on the Camel versions? https://github.com/jbosstools/jbosstools-fuse/pull/1138#discussion_r180427333

apupier replied:
the method updateDependencies is reported as not used, so we can remove it and seems the version is useless :-)

bfitzpat replied:
Version was lifted from one of the original examples I found and am unable to find the same version in any examples today. I'm open to suggestions on how to remedy it.

apupier replied:
ok as it is working with this version currently. I think that we can merge the first iteration as-is on this point.
I created an issue to investigate if we need to have a version that needs to be modifiable across camel versions and/or if we need to use a different one: https://issues.jboss.org/browse/FUSETOOLS-2971

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/WsdlValidator.java:
```diff
@@ -0,0 +1,57 @@
+		try {
+			final URL url = new URL(urlText);
+			InputStream testStream = url.openStream();
+			testStream.close();
```
I guess the testStream needs to be closed in a finally block or to use a try-with-resources https://github.com/jbosstools/jbosstools-fuse/pull/1138#discussion_r180447196

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/WsdlValidator.java:
```diff
@@ -0,0 +1,57 @@
+	/**
+	 * Test to ensure that the URL is accessible.
+	 * @param urlText
+	 * @return
```
might worth mentioning that the return value is HttpStatus.SC_OK in case of success and -1 otherwise.
Or to change it by returning a boolean as the name isURLAccessible of the method might better suggest 
https://github.com/jbosstools/jbosstools-fuse/pull/1138#discussion_r180447925

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/WsdlValidator.java:
```diff
@@ -0,0 +1,57 @@
+			return ValidationStatus.error(UIMessages.wsdl2RestWizardFirstPageValidatorWSDLUrlRequired);
+		}
+		int responseCode = isURLAccessible((String) value);
+		if (responseCode != 200) {
```
reuse HttpStatus.SC_OK ?
or change to boolean? 

https://github.com/jbosstools/jbosstools-fuse/pull/1138#discussion_r180448175

apupier commented on editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java:
```diff
@@ -0,0 +1,359 @@
+	 */
+	protected IProject getSelectedProjectFromSelectionService() {
+		ISelectionService ss = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getSelectionService();
+		String projExpID = "org.eclipse.ui.navigator.ProjectExplorer"; //$NON-NLS-1$
```
even better org.eclipse.ui.IPageLayout.ID_PROJECT_EXPLORER which doesn't require to add another dependency

apupier commented:
I broke test for CamelIOHandlerIT test, works fine locally. test is working fine locally when launched in Eclipse

investigation in progress...

bfitzpat commented:
* Updated warning issue when user selects project (and no project is selected in the project explorer)
* Squashed commits without JIRA issue in commit description (original contribution)
* Removed realm from DataBindingContext definition as suggested
* Still exploring running created projects and have asked @apupier for additional context for how he is running them for Spring and Spring Boot to test

apupier commented:
> Still exploring running created projects and have asked @apupier for additional context for how he is running them for Spring and Spring Boot to test

I cannot reproduce errors from yesterday, so would be fine
